### PR TITLE
debugging: faq: go <=> c migration

### PR DESCRIPTION
**Pull request by PettitWesley**
Signed-off-by: Wesley Pettit <wppttt@amazon.com>

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by ShelbyZ — APPROVED** [review]

File: troubleshooting/debugging.md [hunk 1]
```diff
@@ -893,6 +894,74 @@ However, these plugins were not as performant as the core Fluent Bit C plugins f
...
+    * `net.dns.resolver`
+    * `net.keepalive`
+    * `net.keepalive_idle_timeout`
+    * `net.keepalive_max_recycle`
+    * `net.source_address`
+* If you use log group or stream name templating, each plugin has some support for this but the features and config option names are entirely different.
+    * https://github.com/aws/amazon-cloudwatch-logs-for-fluent-bit#templating-log-group-and-stream-names
+    * https://docs.fluentbit.io/manual/pipeline/outputs/cloudwatch#log-stream-and-group-name-templating-using-record_accessor-syntax
+    * With `cloudwatch` you can put `$()` template variables in the `log_group_name` and `log_stream_name` options. You can then use `default_log_group_name` and `default_log_stream_name` as fallback names if templating fails.
+        * Only `cloudwatch` supports direct templating of ECS metadata when you run in ECS: `$(ecs_task_id)`, `$(ecs_cluster `or `$(ecs_task_arn)`. With `cloudwatch_logs` you can only inject values from the log JSONs. If you want to use ECS Metadata in your config with `cloudwatch_logs` please see: https://github.com/aws-samples/amazon-ecs-firelens-examples/tree/mainline/examples/fluent-bit/init-metadata
```
Comment (matthewfala) [inline troubleshooting/debugging.md:919]: `$(ecs_cluster ` should be `$(ecs_cluster)`

**Review by matthewfala — APPROVED** [review]
Approving, but found a minor typo.

File: troubleshooting/debugging.md [hunk 2]
```diff
@@ -893,6 +894,74 @@ However, these plugins were not as performant as the core Fluent Bit C plugins f
...
+    * With `cloudwatch_logs` templates go in the `log_group_template` or `log_stream_template` and use a `$var` syntax (see doc). Fallback names if templating fails go in the `log_group_name`, `log_stream_name`, or` log_stream_prefix` options.
```
[the remark this answers is not in the document]
Comment (PettitWesley) [reply troubleshooting/debugging.md:920]: oh yea... thanks

[the remark this answers is not in the document]
Comment (PettitWesley) [reply troubleshooting/debugging.md]: addressed

**Pull request merged**